packages/runtime/container-runtime/src/test/gc/gcSummaryStateTracker.spec.ts

```diff
@@ -13,208 +13,269 @@ import {
-				const summary = summaryStateTracker.summarize(
-					false /* fullTree */,
-					true /* trackState */,
-					initialGCState,
-					initialDeletedNodes,
-					newTombstones,
-				);
-				assert(summary?.summary.type === SummaryType.Tree, "GC summary should be a tree");
-				assert(
-					summary.summary.tree[gcStateBlobKey].type === SummaryType.Handle,
-					"GC state should be written as handle",
-				);
-				assert(
-					summary.summary.tree[gcTombstoneBlobKey].type === SummaryType.Blob,
-					"Tombstone state should be written as a blob",
-				);
-				assert(
-					summary.summary.tree[gcDeletedBlobKey].type === SummaryType.Handle,
-					"Deleted nodes should be written as handle",
-				);
-			});
+	it("updates state updated data store count correctly", async () => {
+		const summaryStateTracker = new GCSummaryStateTracker(
+			{
+				shouldRunGC: true,
+				tombstoneMode: true,
+				gcVersionInBaseSnapshot: currentGCVersion,
+				gcVersionInEffect: currentGCVersion,
+			},
+			false /* wasGCRunInBaseSnapshot */,
+		);
 
-			it("does incremental summary when only deleted nodes change", async () => {
-				// Summarize with the same GC state and tombstone state but different deleted nodes as in the initial.
-				// state. The deleted nodes should be summarized as a summary handle.
-				const newDeletedNodes: Set<string> = new Set(...initialDeletedNodes, nodes[2]);
-				const summary = summaryStateTracker.summarize(
-					false /* fullTree */,
-					true /* trackState */,
-					initialGCState,
-					newDeletedNodes,
-					initialTombstones,
-				);
-				assert(summary?.summary.type === SummaryType.Tree, "GC summary should be a tree");
-				assert(
-					summary.summary.tree[gcStateBlobKey].type === SummaryType.Handle,
-					"GC state should be written as handle",
-				);
-				assert(
-					summary.summary.tree[gcTombstoneBlobKey].type === SummaryType.Handle,
-					"Tombstone state should be written as handle",
-				);
-				assert(
-					summary.summary.tree[gcDeletedBlobKey].type === SummaryType.Blob,
-					"Deleted nodes should be written as a blob",
-				);
-			});
-		});
+		// Update the count of data stores whose state updated. This happens when GC runs.
+		summaryStateTracker.updateStateUpdatedDSCount(10);
+		// Validate that the count is 10.
+		assert.strictEqual(
+			summaryStateTracker.stateUpdatedDSCountSinceLastSummary,
+			10,
+			"Updated DS count is not correct",
+		);
+
+		// Call summarize. This happens after GC runs.
+		summaryStateTracker.summarize(
+			false /* fullTree */,
+			true /* trackState */,
+			{ gcNodes: {} },
+			new Set(),
+			[],
+		);
+
+		// Update the count again and validate that the count is aggregate of previous and this count.
+		summaryStateTracker.updateStateUpdatedDSCount(10);
+		assert.strictEqual(
+			summaryStateTracker.stateUpdatedDSCountSinceLastSummary,
+			20,
+			"Updated DS count should be aggregated",
+		);
+
+		// Call summarize again. This happens after GC runs.
```

Comment:
"This happens after GC runs" doesn't seem like the right comment.  Maybe ", as if the first one failed so we need to keep tracking UpdatedDsCount"  (you can find better wording probably)